cmd/migration-managerd/internal/api/workers.go

```diff
@@ -539,40 +662,54 @@ func (d *Daemon) ensureISOImagesExistInStoragePool(ctx context.Context, instance
 	return nil
 }
 
-func (d *Daemon) spinUpMigrationEnv(ctx context.Context, inst migration.Instance, project string, storagePool string) {
+func (d *Daemon) spinUpMigrationEnv(ctx context.Context, inst migration.Instance, project string, storagePool string) error {
 	log := slog.With(
 		slog.String("method", "spinUpMigrationEnv"),
 		slog.String("instance", inst.InventoryPath),
 	)
 
-	// Update the instance status.
-	_, err := d.instance.UpdateStatusByUUID(ctx, inst.UUID, api.MIGRATIONSTATUS_CREATING, api.MIGRATIONSTATUS_CREATING.String(), true)
-	if err != nil {
-		log.Warn("Failed to update instance status", slog.Any("status", api.MIGRATIONSTATUS_ERROR), logger.Err(err))
-		return
-	}
+	var err error
+	reverter := revert.New()
+	defer reverter.Fail()
+	reverter.Add(func() {
+		var errString string
+		if err != nil {
+			errString = err.Error()
+		}
 
-	s, err := d.source.GetByID(ctx, inst.SourceID)
-	if err != nil {
-		log.Warn("Failed to get source by ID", logger.Err(err))
-		_, err = d.instance.UpdateStatusByUUID(ctx, inst.UUID, api.MIGRATIONSTATUS_ERROR, err.Error(), true)
+		err := transaction.Do(ctx, func(ctx context.Context) error {
+			_, err := d.instance.UpdateStatusByUUID(ctx, inst.UUID, api.MIGRATIONSTATUS_ERROR, errString, true)
+			return err
+		})
 		if err != nil {
-			log.Warn("Failed to update instance status", slog.Any("status", api.MIGRATIONSTATUS_ERROR), logger.Err(err))
+			log.Error("Failed to update instance status", slog.Any("status", api.MIGRATIONSTATUS_ERROR.String()), logger.Err(err))
 		}
+	})
```

Comment:
Single statement, no need to wrap it in a transaction.

Reply:
This is a best-effort attempt in the reverter, so I think there's no issue with removing the transactions.